src/NodaTime/Xml/XmlSchemaDefinition.cs

```diff
@@ -0,0 +1,238 @@
+            foreach (var xmlSerializableType in xmlSerializableTypes)
+            {
+                var schemaProvider = xmlSerializableType.GetCustomAttribute<XmlSchemaProviderAttribute>();
+                if (schemaProvider == null)
+                    throw new InvalidOperationException($"The type '{xmlSerializableType}' is missing the '{nameof(XmlSchemaProviderAttribute)}' on its class definition.");
+                var schemaProviderMethod = xmlSerializableType.GetMethod(schemaProvider.MethodName, BindingFlags.Public | BindingFlags.Static);
+                if (schemaProviderMethod == null)
+                    throw new MissingMethodException(xmlSerializableType.Name, schemaProvider.MethodName);
+                schemaProviderMethod.Invoke(null, new object[] { schemaSetForCollecting });
+            }
+            var xmlSchema = new XmlSchema
+            {
+                TargetNamespace = NodaTimeXmlNamespace.Namespace,
+                Namespaces = new XmlSerializerNamespaces(new[] { NodaTimeXmlNamespace })
+            };
+            var schemaTypes = schemaSetForCollecting.Schemas().Cast<XmlSchema>().SelectMany(e => e.Items.OfType<XmlSchemaType>());
+            foreach (var schemaType in schemaTypes.OrderBy(e => e.Name, StringComparer.Ordinal))
+            {
+                var schemaContainsType = xmlSchema.Items.OfType<XmlSchemaType>().Select(e => e.QualifiedName).Contains(schemaType.QualifiedName);
+                if (!schemaContainsType)
+                {
+                    xmlSchema.Items.Add(schemaType);
+                }
+            }
+            var schemaSetForCompiling = new XmlSchemaSet();
+            schemaSetForCompiling.Add(xmlSchema);
+            schemaSetForCompiling.Compile();
+            return xmlSchema;
+        }
+
+        // See https://stackoverflow.com/questions/626319/add-attributes-to-a-simpletype-or-restriction-to-a-complextype-in-xml-schema/626385#626385
+        private static XmlSchemaComplexType CreateSchemaTypeWithAttributes<T>(XmlSchemaType baseType, params XmlSchemaAttribute[] attributes) where T : IXmlSerializable
+        {
+            var content = new XmlSchemaSimpleContentExtension { BaseTypeName = baseType.QualifiedName };
+            foreach (var attribute in attributes)
+            {
+                content.Attributes.Add(attribute);
+            }
+            return new XmlSchemaComplexType
+            {
+                Name = typeof(T).Name,
+                ContentModel = new XmlSchemaSimpleContent { Content = content }
+            };
+        }
+
+        private static XmlSchemaSimpleType CreatePatternRestriction<T>(XmlSchemaType baseType, string pattern) where T : IXmlSerializable
+        {
+            return CreatePatternRestriction(typeof(T).Name, baseType, pattern);
+        }
+
+        private static XmlSchemaSimpleType CreatePatternRestriction(string name, XmlSchemaType baseType, string pattern)
```

Comment:
And again, an expression-bodied member?

Reply:
Done in 4b690b0312668f78e728ff0dc712caea1588f16a.